### PR TITLE
ITS-Study: create pt resolution, add efficiency of sec study

### PR DESCRIPTION
**Pull request by Roberta-Ferioli**
@mconcas

### CODE REVIEW

**Review by mconcas — DISMISSED** [review]
Approving just for CI, do nt merge yet.

**Review by mconcas — APPROVED** [review]

File: Detectors/ITSMFT/ITS/postprocessing/studies/src/TrackCheck.cxx [hunk 1]
```diff
@@ -162,13 +252,170 @@ void TrackCheckStudy::init(InitContext& ic)
...
+  legends.resize(4);
+  legends1Fake.resize(4);
+  legends2Fake.resize(4);
+  legends3Fake.resize(4);
+
+  for (int iH{4}; iH < 8; ++iH) {
+    // check distributions on layers of fake clusters for tracks of different lengths.
+    // Different histograms if the correct cluster exist or not
+    for (int jj = 0; jj < 3; jj++) {
+      histLength[iH - 4][jj] = new TH1I(Form("trk_len_%d_%s", iH, name[jj]), Form("#exists cluster %s", name), 7, -.5, 6.5);
```
Comment (mconcas) [inline Detectors/ITSMFT/ITS/postprocessing/studies/src/TrackCheck.cxx]:
```suggestion
      histLength[iH - 4][jj] = new TH1I(Form("trk_len_%d_%s", iH, name[jj]), Form("#exists cluster %s", name[jj]), 7, -.5, 6.5);
```

File: Detectors/ITSMFT/ITS/postprocessing/studies/src/TrackCheck.cxx [hunk 2]
```diff
@@ -162,13 +252,170 @@ void TrackCheckStudy::init(InitContext& ic)
...
+      histLength[iH - 4][jj]->SetFillColor(colorArr[jj] - 9);
+      histLength[iH - 4][jj]->SetLineColor(colorArr[jj] - 9);
+      histLengthNoCl[iH - 4][jj] = new TH1I(Form("trk_len_%d_nocl_%s", iH, name[jj]), Form("#slash{#exists} cluster %s", name), 7, -.5, 6.5);
```
Comment (mconcas) [inline Detectors/ITSMFT/ITS/postprocessing/studies/src/TrackCheck.cxx]:
```suggestion
      histLengthNoCl[iH - 4][jj] = new TH1I(Form("trk_len_%d_nocl_%s", iH, name[jj]), Form("#slash{#exists} cluster %s", name[jj]), 7, -.5, 6.5);
```

**Review by mconcas — CHANGES_REQUESTED** [review]

File: Detectors/ITSMFT/ITS/postprocessing/studies/src/TrackCheck.cxx [hunk 3]
```diff
@@ -324,73 +863,108 @@ void TrackCheckStudy::updateTimeDependentParams(ProcessingContext& pc)
...
+    setEfficiencyGraph(mEffGoodZ[aa], Form("Good_Z_%s", particleToanalize[aa]), ";Z_{sv} [cm];efficiency secondary particle", colorArr[aa]);
+    fout.WriteTObject(mEffGoodZ[aa].get());
+
+    setEfficiencyGraph(mEffGoodZ[aa], Form("Fake_Z_%s", particleToanalize[aa]), ";Z_{sv} [cm];efficiency secondary particle", colorArr[aa] - 9);
+    fout.WriteTObject(mEffGoodZ[aa].get());
+
+    for (int bb = 0; bb < 4; bb++) {
+      setEfficiencyGraph(mEffGoodPts[aa][bb], Form("EffPtGood_%sl%d", particleToanalize[aa], bb), Form("Good Sec Tracks_%s, L%d"
+                                                                                                       ";#it{p}_{T} (GeV/#it{c});efficiency secondary particle ",
+                                                                                                       bb, particleToanalize[aa]),
```
Comment (mconcas) [inline Detectors/ITSMFT/ITS/postprocessing/studies/src/TrackCheck.cxx]:
```suggestion
                                                                                                       particleToanalize[aa], bb),
```

File: Detectors/ITSMFT/ITS/postprocessing/studies/src/TrackCheck.cxx [hunk 4]
```diff
@@ -324,73 +863,108 @@ void TrackCheckStudy::updateTimeDependentParams(ProcessingContext& pc)
...
+                         colorArr[aa]);
+      setEfficiencyGraph(mEffFakePts[aa][bb], Form("EffPtFake_%sl%d", particleToanalize[aa], bb), Form("Fake Sec Tracks_%s, L%d"
+                                                                                                       ";#it{p}_{T} (GeV/#it{c});efficiency secondary particle ",
+                                                                                                       bb, particleToanalize[aa]),
```
Comment (mconcas) [inline Detectors/ITSMFT/ITS/postprocessing/studies/src/TrackCheck.cxx]:
```suggestion
                                                                                                       particleToanalize[aa], bb),
```

File: Detectors/ITSMFT/ITS/postprocessing/studies/src/TrackCheck.cxx [hunk 5]
```diff
@@ -324,73 +863,108 @@ void TrackCheckStudy::updateTimeDependentParams(ProcessingContext& pc)
...
+                         colorArr[aa]);
+      setEfficiencyGraph(mEffGoodEtas[aa][bb], Form("EffEtaGood_%sl%d", particleToanalize[aa], bb), Form("Good Sec Tracks_%s, L%d"
+                                                                                                         ";#eta ;efficiency secondary particle ",
+                                                                                                         bb, particleToanalize[aa]),
```
Comment (mconcas) [inline Detectors/ITSMFT/ITS/postprocessing/studies/src/TrackCheck.cxx]:
```suggestion
                                                                                                         particleToanalize[aa], bb),
```

File: Detectors/ITSMFT/ITS/postprocessing/studies/src/TrackCheck.cxx [hunk 6]
```diff
@@ -324,73 +863,108 @@ void TrackCheckStudy::updateTimeDependentParams(ProcessingContext& pc)
...
+                         colorArr[aa]);
+      setEfficiencyGraph(mEffFakeEtas[aa][bb], Form("EffEtaFake_%sl%d", particleToanalize[aa], bb), Form("Fake Sec Tracks_%s, L%d"
+                                                                                                         ";#eta ;efficiency secondary particle ",
+                                                                                                         bb, particleToanalize[aa]),
```
Comment (mconcas) [inline Detectors/ITSMFT/ITS/postprocessing/studies/src/TrackCheck.cxx]:
```suggestion
                                                                                                         particleToanalize[aa], bb),
```

**Review by mconcas — CHANGES_REQUESTED** [review]

**Review by mconcas — DISMISSED** [review]

File: Detectors/ITSMFT/ITS/postprocessing/studies/src/TrackCheck.cxx [hunk 7]
```diff
@@ -262,43 +532,242 @@ void TrackCheckStudy::process()
...
+  int evID = 0;
+  int trackID = 0;
+  int totP{0}, goodP{0}, fakeP{0};
   // Currently process only sourceID = 0, to be extended later if needed
   for (auto& evInfo : mParticleInfo[0]) {
+    trackID = 0;
     for (auto& part : evInfo) {
-      if ((part.clusters & 0x7f) != mMask) {
+
+      if (ProcessName[part.prodProcess] == " ")
```
Comment (mconcas) [inline Detectors/ITSMFT/ITS/postprocessing/studies/src/TrackCheck.cxx]:
```suggestion
      if (strcmp(ProcessName[part.prodProcess], " "))
```

**Review by mconcas — APPROVED** [review]

**Comment by mconcas** [comment]
Hi @TimoWilken, I see CentOS 8 builders failing the git fetch on multiple PRs. Will it run periodically so I can expect to have it going through at some point or shall I re-trigger the builds somehow? Thanks in advance.

**Comment by ktf** [comment]
@mconcas @TimoWilken it's not clear to me how this could have passed the fullCI, given it breaks the code checker but apparently the test was green. Do we have some race condition in the tests? Was it force merged?

**Comment by ktf** [comment]
Regression in code checker hopefully fixed by #12094.

**Comment by mconcas** [comment]
> @mconcas @TimoWilken it's not clear to me how this could have passed the fullCI, given it breaks the code checker but apparently the test was green. Do we have some race condition in the tests? Was it force merged?

PR was merged with macos red as it always is lately, fullCI and co were green.
Similar issue in this open pr: https://github.com/AliceO2Group/AliceO2/pull/12044#discussion_r1361857734

Thanks for the fixes, I admittedly relied on automatic checks for this kind of problems.

**Comment by ktf** [comment]
> PR was merged with macos red as it always is lately, fullCI and co were green.

Yes, that's exactly my point. fullCI should have caught it already when checking the PR.

**Pull request merged**